### DISCUSSION TITLE
How to send a message in a specific channel in "guildMemberRemove" event ?

### DISCUSSION THREAD

**Discussion by Izuki13** [discordjs/discord.js#8780] [Q&A] [answered]
My goal is to send a message in a specific channel when a member leaves the server.

I used the method provided by the discord.js documentation :
const channel = client.channels.cache.get('id');
channel. send('content');

The error is always this:
TypeError: Cannot read properties of undefined (reading 'send')

Do you have an explanation of possible errors?

**Comment by almostSouji** [comment, marked as answer]
The error you receive means that `channel` has the value `undefined` which means the cache-lookup `.cache.get(key)` was unsuccessful.
This means that at the time this code runs there is no channel with the provided key in the cache.

Now this, in turn, means that either the key is wrong (make sure you pasted the right value) or that the cache simply does not hold the expected channel. Since channels generally arrive on ready with the initial guild payloads and are updated via incoming events, that suggests the cache might just be empty.

If not already the case, move the code into some execution scope that runs after the client is ready. In your case "when a member leaves the server" would suggest the[ `guildMemberRemove` event](https://discord.js.org/#/docs/discord.js/main/class/Client?scrollTo=e-guildMemberRemove).

If you are retrieving the key from storage (a database/web API), please make sure it is never treated as a number (store as a string, process as a string), as that would truncate the value and corrupt the key. (JavaScript simply cannot deal with numbers as large as Discord ids)
